bin/gwdetchar-omega

```diff
@@ -158,6 +159,40 @@ class OmegaChannelList(object):
 
 # -- Utilities ----------------------------------------------------------------
 
+def check_flag(flag, time, duration, pad):
+    """Check that a state flag is active before processing an OmegaChannel
+
+    Parameters
+    ----------
+    flag : `str`
+        state flag to check
+    time : `float`
+        GPS time of required data
+    duration : `float`
+        duration (in seconds) of required data
+    pad : `float`
+        amount of extra data to read in at the start and end for filtering
+
+    Returns
+    -------
+    check : `bool`
+        Boolean switch to pass (`True`) or fail (`False`) depending on whether
+        the given flag is active
+    """
+    # set GPS start and end time
+    start = time - duration/2. - pad
+    end = time + duration/2. + pad
+    inseg = Segment(start, end)
+    # query for state segments
+    aseg = DataQualityFlag.query(flag, start, end).active
+    # check that state flag is active during the entire analysis
+    if (not aseg.intersects_segment(inseg)) or (len(aseg) > 1):
+        return False
+    elif abs(aseg[0]) < abs(inseg):
```

Comment:
@alurban, I think the `len(aseg) > 1` and `abs(aseg[0]) < abs(inseg)` checks are redundant, so we should only test one.

Reply:
@duncanmmacleod, aha, it seems you're right:

```
[======   ===]
[     =======]
```

In both examples the code wouldn't run because the flag isn't active during the entire segment of interest, and checking that `abs(aseg[0]) < abs(inseg)` is sufficient to establish this. This is reflected in the latest commit.